pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -654,6 +654,16 @@ func DefaultOpenAPIConfig(config configapi.MasterConfig) *openapicommon.Config {
 			},
 		}
 	}
+	if config.OAuthConfig != nil {
+		securityDefinitions["Oauth2AccessToken"] = &spec.SecurityScheme{
+			SecuritySchemeProps: spec.SecuritySchemeProps{
+				Type:             "oauth2",
+				Flow:             "accessCode",
+				AuthorizationURL: config.OAuthConfig.MasterURL + "/oauth/authorize",
```

Comment:
> Why the script should start a server ?

`hack/update-generated-swagger-spec.sh` must build the swagger spec from a running server (there is no other way to get its final runtime state).  The script already uses 127.0.0.1 so you are good to use the variables.